packages/core/src/sessionManagement.ts

```diff
@@ -124,7 +125,12 @@ function retrieveSession(sessionCookie: CookieCache): SessionState {
 }
 
 export function persistSession(session: SessionState, cookie: CookieCache, withNewSessionStrategy = false) {
-  if (withNewSessionStrategy && !utils.isEmptyObject(session)) {
+  if (utils.isEmptyObject(session)) {
+    // clear session
+    cookie.set('', 0)
+    return
+  }
```

Comment:
Thanks, now it's clearer to me :) But I think it would make things even clearer to have a separate function `clearSession`.